core/controllers/admin.py

```diff
@@ -127,6 +138,30 @@ def get(self):
     @require_super_admin
     def post(self):
         """Handles POST requests."""
+        def check_and_update_config_role(old_list, new_list, role):
```

Comment:
There isn't a strict rule against it, but I think having a separate function is better for readability/comprehension.

OK.